packages/ffe-radio-button-react/src/RadioButtonGroup.js

```diff
@@ -0,0 +1,80 @@
+import React, { Component } from 'react';
+import { bool, func, oneOf, shape, string } from 'prop-types';
+import { Tooltip } from '@sb1/ffe-form-react';
+
+class RadioButtonGroup extends Component {
+    render() {
+        const {
+            'aria-invalid': ariaInvalid,
+            children,
+            inline,
+            description,
+            name,
+            onChange,
+            selectedValue,
+            tooltip,
+            tooltipProps,
+        } = this.props;
+
+        const buttonProps = { 
+            'aria-invalid': ariaInvalid, 
+            inline,
+            name,
+            onChange,
+            selectedValue,
+        };
+
+        return (
+            <fieldset 
+                className="ffe-fieldset" 
+                onChange={this.onChange}
+            >
+                {description && 
+                    <legend className="ffe-form-label">
+                        {description}
+                        {tooltip &&
+                            <Tooltip {...tooltipProps}>{tooltip}</Tooltip>
+                        }
+                    </legend>
+                }
+                {children(buttonProps)}
```

Comment:
I approve this pattern :rocket: So much more flexible for wrappers like this.

Reply:
I mentioned this in a [thread on Twitter](https://twitter.com/selbekk/status/969960538003427328) to Kent Dodds - the guy that popularized this render prop pattern. He said

> If the state you want to share is implicit then compound components are great. Otherwise render props are much more powerful and flexible.

I think we want render props for this. The [react docs even have their own section pm them](https://reactjs.org/docs/render-props.html) if you want to learn more.

Reply:
For consistency could we also have the option for single child use the "old fashioned way" without render prop, [like InputGroup](https://github.com/SpareBank1/designsystem/blob/d6baf4bb982c1bc1e2d98b72f1dac315a4a5d68e/packages/ffe-form-react/src/InputGroup.js#L27)?

Reply:
I'm actually thinking no. In my opinion, allowing for both ways was wasn't a good solution. It doubles the API, which makes it hard to understand (and even harder to explain) when to use what. Sure it's less boilerplate, but it's also more of a "hm what should I do here" and "aww crap I have to switch from one to the other" moment prone technique.

If there are very strong opinions on this I'll budge, but I kind of think this is the way to go.

Reply:
> It doubles the API, which makes it hard to understand (and even harder to explain) when to use what.

One child - compose as you normally would. Two or more children - use the render prop. I think that's simple enough to explain. Coupled with a helpful error message when trying to compose as normal with two or more children, I don't see the problem as a user. It's worked very well for `InputGroup` in my opinion.

> it's also more of a "hm what should I do here"

Having the thing break when I try to compose as normal with a single child is also a bit of a "hm what should I do here" :smile: 

> it's also more of a (...) "aww crap I have to switch from one to the other" moment prone technique

True, and it's a bit annoying - I'll give you that. Still, I prefer reducing the amount of boilerplate for the `RadioSwitch` case over always using the render prop. It's not a complex rewrite going from one pattern to the other.

Reply:
@selbekk and I discussed this and he has me convinced that render prop only is the way to go. It avoids the case where a consumer just wraps the two or more children in a `Fragment` or whatever to make the error go away. Less ways to mess things up :+1:

Reply:
I love these discussions! 

Reality is, there was a lot of confusion when I introduced this new technique for input groups. People did what people do and basically tried to hack around the new API by providing a wrapping DIV (which then received the `aria-invalid` and `id` props, breaking the intended effect of having this component in the first place.

I agree that it's... nice to have, if you know the API or read the docs extensively, but I think an API with fewer options is better than a "simple-to-use, simple-to-abuse" API.

Finally, the only use case for a single child here is when you want a `<RadioSwitch />` in there - and we (or the consumer) can build a wrapper around that use case, simplifying its general use case.